### DISCUSSION TITLE
Raspberry Pi Compute Module 5 with eMMC drive

### DISCUSSION THREAD

**Discussion by cms66** [gitbls/sdm#284] [Q&A]
Hi,

I've just got a Compute Module 5 with 32GB eMMC drive (not my preference, but no eMMC option wasn't in stock). I planned to setup boot sequence as NVMe/SD but it seems you can't use SD slot with on board eMMC. I therefore used SDM to build an NVMe drive that works.

The limited instructions I've found for burning eMMC seem more complex than burning an SD or NVMe drive with RPi Imager or SDM. 

When I run command lsblk I get this
NAME         MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
mmcblk0      179:0    0  29.1G  0 disk
mmcblk0boot0 179:32   0     4M  1 disk
mmcblk0boot1 179:64   0     4M  1 disk
nvme0n1      259:0    0 238.5G  0 disk
├─nvme0n1p1  259:1    0   512M  0 part /boot/firmware
└─nvme0n1p2  259:2    0   238G  0 part /

My question is, as the eMMC drive shows up as mmcblk0 (same as SD would), can I burn an image to it with SDM?

Any advice appreciated on this before I potentially break my new toy,

**Comment by gitbls** [comment]
We certainly don't want to break your new toy, er tool 🤣
Unfortunately, I don't have a CM5 (or any other CM for that matter) so haven't looked at the details of burning one. 

If you can provide pointers to the details that you've found I'll be happy to have a look at them and provide some guidance from the sdm perspective.

**Comment by cms66** [comment]
Thanks for the rapid response. I've been looking at
https://www.raspberrypi.com/documentation/computers/compute-module.html
https://www.waveshare.com/wiki/Compute_Module_Burn_EMMC

At least I know I have a working system on NVMe, but having a second
working boot option would be great

**Reply by gitbls** [reply]
Looks to me like you would do the following:
* Linux host
  * Customize an IMG the way you want it to be
  * Configure a host device (preferably a Pi) with rpiboot
  * Connect the host device and the CM however you do that
  * Use sdm on the host device to burn to the newly-discovered device after it shows up

* Windows or Mac host
  * Hope you're not using one of these bc I can't see the directions for these hosts (Doc page is busted in two browsers that I tried)
